testing/unit/rpc/test_transaction.py

```diff
@@ -200,6 +201,178 @@
     assert expected_error_message == error.value.message
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize('ids', [[1], ['123'], 123, True, 'e'])
+async def test_list_batches_with_invalid_ids(request_, ids):
+    """
+    Case: list batches with invalid ids.
+    Expect: exception with invalid id message is raised.
+    """
+    request_.params = {
+        'ids': ids,
+    }
+
+    with pytest.raises(RpcInvalidParamsError) as error:
+        await list_batches(request_)
+
+    assert 'Invalid id.' == error.value.message
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('start', [1, '1', True])
+async def test_list_batches_with_invalid_start(request_, start):
+    """
+    Case: list batches with invalid start field.
+    Expect: exception with invalid id message is raised.
+    """
+    request_.params = {
+        'start': start,
+    }
+
+    with pytest.raises(RpcInvalidParamsError) as error:
+        await list_batches(request_)
+
+    assert 'Invalid id.' == error.value.message
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('limit', [-1, '1', True])
+async def test_list_batches_with_invalid_limit(request_, limit):
+    """
+    Case: list batches with invalid limit field.
+    Expect: exception with invalid limit count message is raised.
+    """
+    request_.params = {
+        'limit': limit,
+    }
+
+    with pytest.raises(RpcInvalidParamsError) as error:
+        await list_batches(request_)
+
+    assert 'Invalid limit count.' == error.value.message
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('head', [1, '1', True])
+async def test_list_batches_with_invalid_head(request_, head):
+    """
+    Case: list batches with invalid start field.
+    Expect: exception with invalid id message is raised.
+    """
+    request_.params = {
+        'head': head,
+    }
+
+    with pytest.raises(RpcInvalidParamsError) as error:
+        await list_batches(request_)
+
+    assert 'Invalid id.' == error.value.message
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('reverse', [1, '1', [1]])
+async def test_list_batches_with_invalid_reverse(request_, reverse):
+    """
+    Case: list batches with invalid reverse field.
+    Expect: exception invalid identifier message is raised.
+    """
+    request_.params = {
+        'reverse': reverse
+    }
+
+    with pytest.raises(RpcInvalidParamsError) as error:
+        await list_batches(request_)
+
+    assert 'Incorrect identifier.' == error.value.message
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('start', ['a'*128])
+async def test_list_batches_with_unexisting_start(request_, start):
```

Comment:
```suggestion
async def test_list_batches_with_non_existing_start(request_, start):
```

Reply:
Done.

Reply:
@Art17, `non-existing` more used in real speech, also a more popular case based on just googling. My 2 cents.